apph-back/src/main/java/com/viseo/apph/service/PhotoService.java

```diff
@@ -61,4 +62,18 @@ public List<PhotoResponse> getUserPhotos(long idUser) {
     public String saveWithName(MultipartFile file, String name) throws InvalidFileException, IOException {
         return s3Dao.upload(file, name);
     }
+
+    public PhotoResponse download(long id) {
+        byte[] photoByte = s3Dao.download(id + "");
+        return new PhotoResponse().setData(photoByte);
+    }
+
+    public Photo getPhoto(long id, long idUser) throws FileNotFoundException {
+        Photo photo = photoDao.getPhoto(id);
+        if (idUser == photo.getIdUser()) {
+            return photo;
+        } else {
+            throw new FileNotFoundException();
+        }
```

Comment:
```suggestion
    public Photo getPhoto(long id, long idUser) throws FileNotFoundException {
        Photo photo = photoDao.getPhoto(id);
        if (photo != null) {
            return photo;
        } else {
            throw new FileNotFoundException();
        }
```
`em.find()` te trouvera forcement la photo avec le bon id, s'il ne le trouve pas il renvoi un null, du coup ici tu vas plus te prendre un `NullPointerException` plutôt qu'un `FileNotFoundException`.

Reply:
après avoir vu ta réponse sur l'une des commentaires plus bas, faudrai plutôt que tu rajoute en plus un verification si la photo existe bien avant de vérifier si la photo appartient à l'user.